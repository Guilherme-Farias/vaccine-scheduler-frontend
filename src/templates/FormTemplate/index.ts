export { default } from './FormTemplate';
export * from './FormTemplate';
