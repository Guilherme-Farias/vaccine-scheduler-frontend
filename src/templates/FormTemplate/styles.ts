import styled, { css } from 'styled-components/macro';

export const Container = styled.div`
  margin: auto;
  height: 100%;
  max-width: 420px;
`;

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes['3xl']};
    font-weight: ${theme.font.semiBold};
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xl};
    color: ${theme.colors['gray-500']};
    margin-bottom: ${theme.spacings.xl};
  `}
`;
