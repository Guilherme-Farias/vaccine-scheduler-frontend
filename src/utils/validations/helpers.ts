import Joi from 'joi';
import { IParamErrors } from 'types/global';

export function getFieldErrors(objError: Joi.ValidationResult) {
  const errors: IParamErrors = {};

  if (objError.error) {
    objError.error.details.forEach(err => {
      errors[err.path.join('.')] = err.message;
    });
  }

  return errors;
}
