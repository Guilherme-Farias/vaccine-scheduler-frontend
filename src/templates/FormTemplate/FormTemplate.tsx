import React from 'react';
import MaxWidthWrapper from 'components/MaxWidthWrapper';
import * as S from './styles';

export type FormTemplateProps = {
  title: string;
  description: string;
  children: React.ReactNode;
};

const FormTemplate: React.FC<FormTemplateProps> = ({
  title,
  description,
  children,
}: FormTemplateProps) => (
  <MaxWidthWrapper>
    <S.Container>
      <S.Title>{title}</S.Title>
      <S.Description>{description}</S.Description>
      <section>{children}</section>
    </S.Container>
  </MaxWidthWrapper>
);

export default FormTemplate;
