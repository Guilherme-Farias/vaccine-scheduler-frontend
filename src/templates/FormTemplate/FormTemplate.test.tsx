import { render, screen } from 'utils/tests/helpers';

import FormTemplate from '.';

describe('<FormTemplate />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <FormTemplate title="Title" description="description">
        <form data-testid="test" />
      </FormTemplate>,
    );
    expect(screen.getByRole('heading', { name: /title/i })).toBeInTheDocument();
    expect(screen.getByText(/description/i)).toBeInTheDocument();
    expect(screen.getByTestId(/test/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
