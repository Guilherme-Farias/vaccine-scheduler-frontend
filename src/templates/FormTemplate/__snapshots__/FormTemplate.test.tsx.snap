// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormTemplate /> should render the heading 1`] = `
.c0 {
  width: 100%;
  max-width: 81.25rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(2rem / 2);
  padding-right: calc(2rem / 2);
}

.c1 {
  margin: auto;
  height: 100%;
  max-width: 420px;
}

.c2 {
  font-size: 1.875rem;
  font-weight: 600;
}

.c3 {
  font-size: 1.25rem;
  color: #6b7280;
  margin-bottom: 1.5rem;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h2
      class="c2"
    >
      Title
    </h2>
    <p
      class="c3"
    >
      description
    </p>
    <section>
      <form
        data-testid="test"
      />
    </section>
  </div>
</div>
`;
